### DISCUSSION TITLE
When evaluating the retrieval in the NQ dataset, what is the criteria of correct retrieval?

### DISCUSSION THREAD

**Discussion by miguelwon** [castorini/pyserini#1515] [Q&A] [answered]
Hi,

Looking for some help to clear up my lack of full understanding on the evaluation. In the Metrics section of [A Replication Study of Dense Passage Retriever](https://arxiv.org/pdf/2104.05740.pdf), it says that "we measured effectiveness in terms of top-k retrieval accuracy, defined as the fraction of questions that have a correct answer span in the top-k retrieved contexts at least once." 

What is the definition of a "correct answer"? Or to put it another way, for the NQ dataset how can we determine whether the passages retrieved from the Wikipedia corpus are accurate, considering it's highly improbable that there will be a passage that precisely matches the annotated short or long answer?

Thanks!

**Comment by ronakice** [comment, marked as answer]
Yes, the passage does not precisely have to match the annotated short/long answer. We process passages to find those that contain some variant of one of the gold answers, up to normalization, as an exact match. This set is by no means exhaustive or perfect but it is the best one can do in a compute-friendly way. I hope that clarifies things!

**Reply by miguelwon** [reply]
I see. In that case what is the criteria? Rouge? Do you know where can I find more details? 
Thanks a lot the quick answer!

**Reply by ronakice** [reply]
You can trace [this function call]( https://github.com/castorini/pyserini/blob/87354e6f467e11ba92a20dab73fc7bc0ca992f86/pyserini/eval/convert_trec_run_to_dpr_retrieval_run.py#L72) for more information
